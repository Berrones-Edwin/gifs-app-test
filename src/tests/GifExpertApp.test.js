import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import React from 'react'
import { GifExpertApp } from '../GifExpertApp'

describe('tests in zGifExpertApp />', () => {
  test('should show component successfully', () => {
    const wrapper = shallow(<GifExpertApp />)

    expect(wrapper).toMatchSnapshot()
})

test('should render a list categories', () => {
    const categories = ['random','other categorie']
    const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>)
    
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('GifGrid').length).toBe(categories.length)


  });
  
})
