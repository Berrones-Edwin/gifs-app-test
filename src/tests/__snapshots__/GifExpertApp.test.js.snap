// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in zGifExpertApp /> should render a list categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="random"
      key="random"
    />
    <GifGrid
      category="other categorie"
      key="other categorie"
    />
  </ol>
</Fragment>
`;

exports[`tests in zGifExpertApp /> should show component successfully 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
